src/wp-includes/l10n/class-wp-translation-controller.php

```diff
@@ -0,0 +1,410 @@
+	 * @param string $textdomain       Optional. Text domain. Default 'default'.
+	 * @param string $locale           Optional. Locale. Default current locale.
+	 * @return bool True on success, false otherwise.
+	 */
+	public function load_file( string $translation_file, string $textdomain = 'default', string $locale = null ) {
+		if ( null === $locale ) {
+			$locale = $this->current_locale;
+		}
+
+		$translation_file = realpath( $translation_file );
+
+		if ( false === $translation_file ) {
+			return false;
+		}
+
+		if (
+			isset( $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ] ) &&
+			false !== $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ]
+		) {
+			return null === $this->loaded_files[ $translation_file ][ $locale ][ $textdomain ]->error();
+		}
+
+		if (
+			isset( $this->loaded_files[ $translation_file ][ $locale ] ) &&
+			array() !== $this->loaded_files[ $translation_file ][ $locale ]
+		) {
+			$moe = reset( $this->loaded_files[ $translation_file ][ $locale ] );
+		} else {
+			$moe = WP_Translation_File::create( $translation_file );
+			if ( false === $moe || null !== $moe->error() ) {
+				$moe = false;
+			}
+		}
+
+		$this->loaded_files[ $translation_file ][ $locale ][ $textdomain ] = $moe;
+
+		if ( ! $moe instanceof WP_Translation_File ) {
+			return false;
+		}
+
+		if ( ! isset( $this->loaded_translations[ $locale ][ $textdomain ] ) ) {
+			$this->loaded_translations[ $locale ][ $textdomain ] = array();
+		}
+
+		$this->loaded_translations[ $locale ][ $textdomain ][] = $moe;
+
+		return true;
+	}
+
+	/**
+	 * Unloads a translation file for a given text domain.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Translation_File|string $file       Optional. Translation file instance or file name. Default all files.
+	 * @param string                     $textdomain Optional. Text domain. Default 'default'.
+	 * @param string                     $locale     Optional. Locale. Default all locales.
+	 * @return bool True on success, false otherwise.
+	 */
+	public function unload_file( $file, string $textdomain = 'default', string $locale = null ) {
+		if ( is_string( $file ) ) {
+			$file = realpath( $file );
+		}
+
+		if ( null !== $locale ) {
+			foreach ( $this->loaded_translations[ $locale ][ $textdomain ] as $i => $moe ) {
+				if ( $file === $moe || $file === $moe->get_file() ) {
+					unset( $this->loaded_translations[ $locale ][ $textdomain ][ $i ] );
+					unset( $this->loaded_files[ $moe->get_file() ][ $locale ][ $textdomain ] );
+					return true;
+				}
+			}
+
+			return true;
+		}
+
+		foreach ( $this->loaded_translations as $l => $domains ) {
+			foreach ( $domains[ $textdomain ] as $i => $moe ) {
+				if ( $file === $moe || $file === $moe->get_file() ) {
+					unset( $this->loaded_translations[ $l ][ $textdomain ][ $i ] );
+					unset( $this->loaded_files[ $moe->get_file() ][ $l ][ $textdomain ] );
+					return true;
+				}
+			}
+		}
+
+		return true;
```

Comment:
Was this meant to return `false`?

```suggestion
		return false;
```